R/subscript-tables.R

```diff
@@ -149,5 +159,46 @@ updateTableAttributes <- function(y, x, called.args) {
     names(attributes(y))[names.needing.update] <- paste0("original.", attr.names[names.needing.update])
     attr(y, "name") <- paste0(x.attributes[["name"]], "[",
                               paste(as.character(called.args), collapse = ","), "]")
+    y <- updateQStatisticsTestingInfo(y, x, evaluated.args)
+    y
+}
+
+
+updateQStatisticsTestingInfo <- function(y, x, evaluated.args)
+{
+    q.test.info <- attr(x, "QStatisticsTestingInfo")
+    if (is.null(q.test.info))
+        return(y)
+    z.attr <- as.vector(q.test.info [ , "zstatistic"])
+
+    dim.len <- length(dim(x))
+    is.multi.stat <- is.null(attr(x, "statistic"))
+    stat.names <- if (is.multi.stat){
+        dimnames(x)[[dim.len]]
+    }else
+        attr(x, "statistic")
+
+    if (is.multi.stat)
+    {
+        evaluated.args <- evaluated.args[-length(evaluated.args)]
+        dim.len <- dim.len - 1
+    }
+    qtypes <- attr(x, "questiontypes")
+    grid.types <- c("PickAnyGrid", "PickOneMulti", "NumberGrid")  # "NumberMulti",
+    grid.in.cols <- length(qtypes) > 1 && qtypes[2] %in% grid.types
+    if (grid.in.cols)
+        perm <- switch(dim.len, NaN, 2:1, c(3,1,2), c(4, 2, 1, 3))
+    else
+        perm <- dim.len:1
+
+    idx.dim <- if(!is.multi.stat) dim(x)
+               else dim(x)[-(dim.len + 1)]
+    idx.array <- array(1:length(x), dim = dim(x))
+    q.test.info.idx <- as.vector(aperm(idx.array, perm))
+
+    remaining.idx <- as.vector(do.call(`[`, c(list(idx.array), evaluated.args)))
+
+    q.test.info <- q.test.info[q.test.info.idx %in% remaining.idx, , drop = FALSE]
```

Comment:
Assuming this is the consequence of the 'row-major' storage of the attribute when there are different dim arrays for the combinations of Question types. Might be worth adding some inline comments to aid the unfamiliar reader (one of us later when we forget how this all works).